python_modules/dagster/dagster/core/definitions/event_metadata.py

```diff
@@ -330,19 +332,97 @@ def __new__(cls, asset_key: "AssetKey"):
         )
 
 
+@whitelist_for_serdes
+class TableSchemaMetadataEntryData(NamedTuple("_TableSchemaMetadataEntryData", [("schema", Dict)])):
+    """Representation of a schema for tabular data. Schema must be formatted as a
+    `Frictionless Table Schema<https://specs.frictionlessdata.io//table-schema/>`,
+    with the following modifications:
+
+    - The `type` of each field MAY be an arbitrary string (i.e. it is not restricted
+      to Frictionless types).
+    - Field `constraints` descriptor MAY contain a property `other` which
+      contains an array of strings. Each element should describe a constraint
+      that is not expressible with predefined Frictionless constraint types.
+    - A top-level property `constraints` MAY be included. This value is a
+      descriptor for "table-level" constraints. Presently only one property,
+      `other` is supported. This should contain a list of strings describing
+      arbitrary table-level constraints.
+
+    .. code-block:: python
+
+            # example schema
```

Comment:
This JSON-esque blob looks like a good over-the-wire representation of table metadata, but I think that modeling the structure with Python types would make for a less error-prone experience when users are constructing these.  E.g. if we offer a `Column` class with "name", "description", etc. fields, then static type checkers will be able to tell users when they have a typo in an attr name.